src/reader.rs

```diff
@@ -83,14 +83,18 @@ pub fn find(needle: &ReadUntil, buffer: &str, eof: bool) -> Option<(usize, usize
                 None
             }
         }
-        ReadUntil::Any(ref any) => {
-            for read_until in any {
-                if let Some(pos_tuple) = find(read_until, buffer, eof) {
-                    return Some(pos_tuple);
+        ReadUntil::Any(ref anys) => anys
+            .iter()
+            // Filter matching needles
+            .filter_map(|any| find(any, buffer, eof))
+            // Return the left-most match
+            .min_by(|(start1, end1), (start2, end2)| {
+                if start1 == start2 {
+                    end1.cmp(end2)
+                } else {
+                    start1.cmp(start2)
```

Comment:
You realize that the above change does not result in equal functionality?

If that's desired, the commit message should note why.

Reply:
Yes, that is intentional, something that has previously been non-deterministic, will now be deterministic, on one of the values that the function might have previously returned.